### ISSUE TITLE
Add term-style proof of app_vec_partial

### ISSUE THREAD

**Issue by vaibhavkarve** [vaibhavkarve/igl2020#25]
There is already a tactic-style proof of `app_vec_partial` here:
https://github.com/vaibhavkarve/igl2020/blob/2711656f7f6ccb3c95a19254a49d3889e454afa1/src/model.lean#L83

Given that this is a definition, it should be turned into a term-style proof. The first step of the proof is induction on `m`. This will translate into pattern matching on `m`. To allow pattern matching, the type of the function will have to be changed to
```lean
def app_vec_partial' {α : Type} {n : ℕ} : Π {m : ℕ},
  m ≤ n → Func α (n+1) → vector α (m+1) → Func α (n-m)
| 0 := sorry
| (m+1) := sorry
```
Note also that "induction on m" in the tactic-style proof indicates that this will be a recursive definition when translated to a term-style proof.

**Comment by vaibhavkarve** [comment]
@kthmp I am assigning this to you since you were already working on a tactic-style proof for this. Anyone else who wishes to work on this can aassign themselves to this issue.

**Comment by vaibhavkarve** [comment]
Closed in #73
